### PR TITLE
[sbt-metals] Fix JDK version when Java home is jre folder

### PR DESCRIPTION
**Pull request by adpi2**
My Java home is  `~/.cache/coursier/jvm/adopt@1.8.0-275/jre` and `MetalsPlugin` could not find the associated JDK version.

This fixes the bug:
- try to read the `$javaHome/../release` file if the `$javaHome/release` file does not exist
- try to read the `$javaHome/lib/rt.jar` file if the `$javaHome/jre/lib/rt.jar` does not exist.

### CODE REVIEW

File: sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala [hunk 1]
```diff
@@ -139,48 +139,47 @@ object MetalsPlugin extends AutoPlugin {
...
+      def jdk8Fallback: Option[JdkVersion] =
+        Seq(javaHome / "jre" / "lib" / "rt.jar", javaHome / "lib" / "rt.jar")
+          .filter(_.exists)
+          .map(_ => JdkVersion(8))
+          .headOption
 
       fromReleaseFile.orElse(jdk8Fallback)
     }
 
     def parse(v: String): Option[JdkVersion] = {
```
Comment (ckipp01) [inline sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala:164]: Take it or leave it, but do you think this is easier to read:

```scala
    def parse(v: String): Option[JdkVersion] = {
      val numbers = Try {
        v
          .split('-')
          .head
          .split('.')
          .toList
          .take(2)
          .flatMap(s => Try(s.toInt).toOption)
      }.toOption

      numbers match {
        case Some(1 :: minor :: _) =>
          Some(JdkVersion(minor))
        case Some(single :: _) =>
          Some(JdkVersion(single))
        case _ => None
      }
    }
```
Might simplify it a bit.

**Review by ckipp01 — APPROVED** [review]
Looks good to me! Just a nit, but it might just be a style thing, so no worries if you ignore it.

**Review by dos65 — APPROVED** [review]
LGTM!

Also, it would be nice if you would copy these changes to https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/JavaInteractiveSemanticdb.scala

This code was copypasted from there

File: sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala (hunk 1 above)
Comment (adpi2) [reply sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala:164]: :+1:

Comment (dos65) [reply sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala:164]: btw, I also realized that there is no need in first split by `-`
```
      val numbers = Try {
        v
          .split('.')
          .toList
          .take(2)
          .flatMap(s => Try(s.toInt).toOption)
      }.toOption
```

Comment (adpi2) [reply sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala:164]: The split by `-` is actually needed to parse the Java version `17-ea`, as in this [test](https://github.com/scalameta/metals/blob/3a9ab1d71c3275276e1e0e44f17264fc33d0bfaf/tests/unit/src/test/scala/tests/JavaInteractiveSemanticdbSuite.scala#L10).

Comment (dos65) [reply sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala:164]: oh, yep. sorry for the wrong suggestion!

**Pull request merged**